src/pages/registration/new_registration/UploadFileRegistration.tsx

```diff
@@ -0,0 +1,154 @@
+    mutationFn: async () => {
+      const newRegistration = await createRegistration();
+      return newRegistration.data;
+    },
+    onError: () => dispatch(setNotification({ message: t('feedback.error.create_registration'), variant: 'error' })),
+  });
+
+  const deleteFileMutation = useMutation({
+    mutationFn: async ({ registrationIdentifier, fileIdentifier }: DeleteFileMutationParams) => {
+      if (registrationIdentifier && fileIdentifier) {
+        await deleteRegistrationFile(registrationIdentifier, fileIdentifier);
+      }
+    },
+    onSuccess: () => dispatch(setNotification({ message: t('feedback.success.delete_file'), variant: 'success' })),
+    onError: () => dispatch(setNotification({ message: t('feedback.error.delete_file'), variant: 'error' })),
+  });
+
+  return (
+    <RegistrationAccordion elevation={5} expanded={expanded} onChange={onChange}>
+      <AccordionSummary
+        data-testid={dataTestId.registrationWizard.new.fileAccordion}
+        expandIcon={<ExpandMoreIcon fontSize="large" />}>
+        <CloudUploadIcon />
+        <div>
+          <Typography variant="h2">{t('registration.registration.start_with_uploading_file_title')}</Typography>
+          <Typography>{t('registration.registration.start_with_uploading_file_description')}</Typography>
+        </div>
+      </AccordionSummary>
+
+      <AccordionDetails>
+        {uppy ? (
+          <FileUploader uppy={uppy} addFile={(newFile) => setUploadedFiles((files) => [newFile, ...files])} />
+        ) : (
+          <Button
+            data-testid={dataTestId.registrationWizard.new.uploadFileButton}
+            variant="contained"
+            sx={{ alignSelf: 'center' }}
+            endIcon={<UploadIcon />}
+            loadingPosition="end"
+            loading={createRegistrationMutation.isPending}
+            onClick={() => fileInputRef.current?.click()}>
+            {t('common.select_file')}
+            <input
+              type="file"
+              hidden
+              ref={fileInputRef}
+              onChange={async (event) => {
+                const newRegistration = await createRegistrationMutation.mutateAsync();
+                const newUppyInstance = createUppy(i18n.language, newRegistration.identifier);
+                setUppy(newUppyInstance);
+                const file = event.target.files?.[0];
+                if (file) {
+                  newUppyInstance.addFile(file);
+                }
+              }}
+            />
+          </Button>
+        )}
+
+        {uppy && uploadedFiles.length > 0 && (
+          <>
+            <Typography variant="h3">{t('registration.files_and_license.files')}:</Typography>
+            {uploadedFiles.map((file) => (
+              <Box key={file.identifier} sx={{ display: 'flex', alignItems: 'center', gap: '0.5rem' }}>
+                <Typography sx={{ wordBreak: 'break-all' }}>{file.name}</Typography>
+                <DeleteIconButton
```

Comment:
Vis loading ved sletting

Reply:
Dette er veldig lett med siste versjon av MUI som har en `loading` prop på IconButton også, men det virker dårlig siden vi har lagt til en del overstyring av styling etc i `DeleteIconButton`:
![bilde](https://github.com/user-attachments/assets/1a419d4e-c77e-4bc4-b0dc-15a19d03425c)

Så jeg må gå en runde på den også

Reply:
Endrer til å bruke `IconButton` direkte her, i stedet for vår egne wrapper-komponent `<DeleteIconButton>`. Om jeg skal endre på den ser jeg det må endres mange andre plasser hvor den brukes også. Tror uansett det ikke var så lurt å ha en egen komponent som gjør mye overskriving av styling, da det fort knekker når det skal brukes på en litt annen måte.

Nå vil det se sånn ut:
![bilde](https://github.com/user-attachments/assets/edb0edab-3d04-42a9-b3e0-b2f6388770c4)